docs/reference/semantic-segmentation.md

```diff
@@ -0,0 +1,153 @@
+## semantic_segmentation module
+
+The *semantic segmentation* module contains the *BisenetLearner* class, which inherit from the abstract class *Learner*.
+
+
+### Class BisenetLearner
+Bases: `engine.learners.Learner`
+
+The *BisenetLearner* class is a wrapper of the BiseNet model [[1]](#bisenetp) found on [BiseNet] (https://github.com/ooooverflow/BiSeNet).
+It is used to train Semantic Segmentation models on RGB images and run inference.
+
+
+
+
+
+The [BisenetLearner](#src.opendr.perception.semantic_segmentation.bisenet.bisenet_learner.py) class has the
+following public methods:
+
+#### `BisenetLearner` constructor
+```python
+BisenetLearner(self, lr, iters, batch_size, optimizer, checkpoint_after_iter, checkpoint_load_iter, val_after, temp_path, device, loss, weight_decay, momentum, drop_last, pin_memory, num_workers, num_classes)
+```
+
+Constructor parameters:
+  - **lr**: *float, default=0.01*  
+    Learning rate during optimization. 
+  - **iters**: *int, default=500*  
+    Number of epochs to train for. 
+  - **batch_size**: *int, default=2*  
+    Dataloader batch size. Defaults to 2.
+  - **optimizer**: *str, default="sgd"*  
+    Name of optimizer to use ("sgd" ,"rmsprop", or "adam"). 
+  - **checkpoint_after_iter**: *int, default=0*  
+    Save chackpoint after specific epochs. 
+  - **checkpoint_load_iter**: *int, default=0*  
+    Unused parameter. 
+  - **val_after**: *int, default=0*  
+    Perform validation after specific epochs. 
+  - **temp_path**: *str, default='./temp'*  
+    Path in which to store temporary files. 
+  - **device**: *str, default="cuda"*  
+    Name of computational device ("cpu" or "cuda"). 
+  - **loss_func**: *str, default=torch.nn.CrossEntropyLoss()*  
+    Loss function used during optimization. 
+  - **weight_decay**: *[type], default=1e-5*  
+    Weight decay used for optimization. 
+  - **momentum**: *float, default=0.9*  
+    Momentum used for optimization. 
+  - **drop_last**: *bool, default=True*  
+    Drop last data point if a batch cannot be filled. 
+  - **pin_memory**: *bool, default=False*  
+    Pin memory in dataloader. 
+  - **num_workers**: *int, default=4*  
+    Number of workers in dataloader. 
+  - **num_classes**: *int, default=12*  
+    Number of classes to predict among. 
+  - **crop_height**: *int, default=720*
+    Input image height.
+  - **crop_width**: *int, default=960*
+    Input image width.
+  - **context_path**: *str, default='resnet18'*
+    Context path for the bisenet model.
+
+
+#### `BisenetLearner.fit`
+```python
+BisenetLearner.fit(self, dataset, val_dataset=None, val_dataset=None, logging_path='', silent=False, verbose=True)
```

Comment:
Usually default values are only listed below and not here (same for other methods).
Also note that `val_dataset` is listed twice.